meta/meta.go

```diff
@@ -0,0 +1,119 @@
+package meta
+
+import (
+	"strings"
+
+	"pkg.re/essentialkaos/ek.v9/fmtc"
+	"pkg.re/essentialkaos/ek.v9/fsutil"
+	"pkg.re/essentialkaos/ek.v9/jsonutil"
+	"pkg.re/essentialkaos/ek.v9/path"
+)
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+type VMetadata struct {
+	*VMetadataRepository        // metadata of the repository
+	StorageURL           string // repository URL and port
+	StoragePath          string // repository path to store images and metadata
+}
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+// nameMeta returns name of the metadata file with extension
+func (m *VMetadata) nameMeta() string {
+	return fmtc.Sprintf("%s.json", m.Name)
+}
+
+// baseMeta returns name of metadata files
+func (m *VMetadata) baseMeta() string {
+	return path.Join(m.StoragePath, m.Name)
+}
+
+// DirMeta returns directory string to metadata file
+func (m *VMetadata) DirMeta() string {
+	return path.Join(m.baseMeta(), "metadata")
+}
+
+// PathMeta returns full path to metadata file
+func (m *VMetadata) PathMeta() string {
+	return path.Join(m.DirMeta(), m.nameMeta())
+}
+
+// URLMeta returns direct link to metadata file
+func (m *VMetadata) URLMeta() string {
+	return fmtc.Sprintf("%s/%s/%s/%s",
```

Comment:
Use `fmt` package instead of `fmtc` here. `fmtc` _always_ try to find formatting tags in input and convert it to escape characters.